docs/modules/ROOT/examples/code/druid-ldap-authentication.yaml

```diff
@@ -0,0 +1,43 @@
+---
+apiVersion: druid.stackable.tech/v1alpha1
+kind: DruidCluster
+metadata:
+  name: derby-druid
+spec:
+  clusterConfig:
+    authentication:
+      - authenticationClass: druid-ldap-auth-class
+---
+apiVersion: authentication.stackable.tech/v1alpha1
+kind: AuthenticationClass
+metadata:
+  name: druid-ldap-auth-class
+spec:
+  provider:
+    ldap:
+      hostname: openldap
+      searchBase: ou=users,dc=example,dc=org
+      searchFilter: (uid=%s)
+      port: 1389
+      bindCredentials:
+        secretClass: druid-ldap-secret
+---
+apiVersion: secrets.stackable.tech/v1alpha1
+kind: SecretClass
+metadata:
+  name: druid-ldap-secret
+spec:
+  backend:
+    k8sSearch:
+      searchNamespace:
+        pod: {}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: druid-ldap-secret
+  labels:
+    secrets.stackable.tech/class: druid-ldap-secret
+stringData:
+  user: uid=admin,ou=Users,dc=example,dc=org
+  password: admin
```

Comment:
```suggestion
# yamllint disable-file

# tag::authclass[]
apiVersion: authentication.stackable.tech/v1alpha1
kind: AuthenticationClass
metadata:
  name: ldap-auth
spec:
  [...]
#end::authclass[]

# tag::druid[]
apiVersion: druid.stackable.tech/v1alpha1
kind: DruidCluster
metadata:
  name: druid
spec:
  clusterConfig:
    authentication:
      - authenticationClass: ldap-auth
  [...]
# end::druid[]
```

The existing example is too complete, and contains the Secret and SecretClass, which are just used by the AuthenticationClass. I suggest to leave both of these out (they are covered in the AuthenticationClass docs and tutorial) and only focus on what's important for Druid:
- there is an authentication class with a name
- You need to reference it in a particular place in the DruidCluster resource

In my suggestion I'm using tags in the source file, to include both sections in seperate places. See also [here](https://github.com/stackabletech/trino-operator/blob/main/docs/modules/usage_guide/examples/code/trino-ldap-auth-snippet.yaml) and [here](https://github.com/stackabletech/trino-operator/blob/main/docs/modules/usage_guide/pages/security.adoc)

Reply:
Great point. Thanks for the suggested changes! I added them in separate commits.